src/driver/read_fixed.rs

```diff
@@ -1,27 +1,30 @@
 use crate::buf::fixed::FixedBuf;
-use crate::buf::{IoBuf, IoBufMut, Slice};
+use crate::buf::BoundedBufMut;
 use crate::driver::op::{self, Completable};
 use crate::driver::{Op, SharedFd};
 use crate::BufResult;
 
 use std::io;
 
-pub(crate) struct ReadFixed {
+pub(crate) struct ReadFixed<T> {
     /// Holds a strong ref to the FD, preventing the file from being closed
     /// while the operation is in-flight.
     #[allow(dead_code)]
     fd: SharedFd,
 
     /// The in-flight buffer.
-    buf: Slice<FixedBuf>,
+    buf: T,
 }
 
-impl Op<ReadFixed> {
+impl<T> Op<ReadFixed<T>>
+where
+    T: BoundedBufMut<BufMut = FixedBuf>,
```

Comment:
Interesting, the associated type is FixedBuf. And not a FixedBufMut?

Reply:
There is no `FixedBufMut`, a `FixedBuf` handle can be used as a mutable buffer.